src/main/java/com/scalar/db/api/Operation.java

```diff
@@ -51,6 +79,45 @@ public Operation(Key partitionKey, Key clusteringKey) {
     return tableName;
   }
 
+  /**
+   * Returns the full table name with the full namespace for this operation
+   *
+   * @return an {@code Optional} with the returned the full table name
+   */
+  @Nonnull
+  public Optional<String> forFullTableName() {
+    if (!namespace.isPresent() || !tableName.isPresent()) {
+      return Optional.empty();
+    }
+
+    if (namespacePrefix.isPresent()) {
```

Comment:
I think it is a bit redundant. How about this?
```    
    StringBuilder builder = new StringBuilder();
    if (namespacePrefix.isPresent()) {
      builder.append(namespacePrefix.get()));
    }
    builder.append(namespace.get());
    builder.append(".");
    builder.append(tableName.get()); 
    return Optional.of(builder.toString());
```

Reply:
Thank you! The suggestion looks better. Modifled.